deCODE_by_CHR.py

```diff
@@ -0,0 +1,172 @@
+    filter_conditions = (
+        (hl.is_missing(mt["allele_type"]))
+        | (
+            (hl.is_defined(mt["allele_type"]))
+            & (mt["allele_type"] == "snv")
+            & (mt["AS_VQSLOD"] < mt["filtering_model"].snv_cutoff.min_score)
+        )
+        | (
+            (hl.is_defined(mt["allele_type"]))
+            & (mt["allele_type"] == "ins")
+            & (mt["AS_VQSLOD"] < mt["filtering_model"].indel_cutoff.min_score)
+        )
+        | (hl.len(mt.alleles) != 2)
+        | ((hl.len(mt.alleles) == 2) & (mt.n_unsplit_alleles != 2))
+        | (hl.is_missing(mt["InbreedingCoeff"]))
+        | ((hl.is_defined(mt["InbreedingCoeff"])) & (mt["InbreedingCoeff"] < -0.3))
+    )
+    mt = mt.filter_rows(filter_conditions, keep=False)
+
+    # Restricted to high quality variants (GQ>=20, DP>=10)
+    filter_condition = (mt.DP >= 10) & (mt.GQ >= 20)
+    mt = hl.variant_qc(mt.filter_entries(filter_condition, keep=True))
+
+    """
+    Step 4 - deCODE specific filter
+    1. Exclude variants with call rate < 0.99
+    2. Identify singleton mutations
+    3. Apply deCODE specific filter (DP >= 16, GQ >= 90, >=3 indepedent reads for alt allele, not in simple repeat regions, very low AF at population level[use deCODE sample size as a ref])
+    """
+
+    # Exclude variants with call rate < 0.99 (not in deCODE paper)
+    mt = mt.filter_rows(mt.variant_qc.call_rate >= 0.99)
+
+    # Identify singleton mutations
+    mt = mt.filter_rows(mt.variant_qc.n_non_ref == 1)
+
+    # Apply strict filters to singleton mutations
+    # DP >= 16 & GQ >= 90
+    # >=3 indepedent reads containing a variant allele required
+    filter_condition = (mt.DP >= 16) & (mt.GQ >= 90) & (mt.AD[1] >= 3)
+    mt = hl.variant_qc(mt.filter_entries(filter_condition, keep=True))
+    mt = mt.filter_rows(mt.variant_qc.n_non_ref == 1)
+
+    # Read gnomAD allele frequency
+    ref_ht = hl.read_table(gnomad_file)
+
+    # Annotate variants with CADD scores, gnomAD etc.
+    mt = mt.annotate_rows(
+        cadd=ref_ht[mt.row_key].cadd,
+        gnomad_genomes=ref_ht[mt.row_key].gnomad_genomes,
+        gnomad_genome_coverage=ref_ht[mt.row_key].gnomad_genome_coverage,
+    )
+
+    # Delete gnomAD file to save space
+    del ref_ht
+
+    # Apply gnomAD AF filter (very low MAF)
+    AF_cutoff = 1 / (int(cohort_size) * 2)
+    mt = mt.filter_rows(mt.gnomad_genomes.AF_POPMAX_OR_GLOBAL <= AF_cutoff)
+
+    # Exclude mutations in simple repeat regions
+    # simple repeat regions - combining the entire Simple Tandem Repeats by TRF track in UCSC hg38 with all homopolymer regions in hg38 of length 6bp or more
+
+    # Read the (Combined) Simple Repeat Regions
+    interval_table = hl.import_bed(regions_file, reference_genome="GRCh38")
+
+    # Exclude mutations in these regions
+    mt = hl.variant_qc(
+        mt.filter_rows(hl.is_defined(interval_table[mt.locus]), keep=False)
+    )
+
+    """
+    Step 5 - Export to a pVCF file
```

Comment:
How large do you expect this to be? Will this export all samples? VCFs aren't great for thousands of samples -- are you sure you can't just keep this as a Hail MatrixTable, potentially with all unnecessary fields removed?

Reply:
To be honest, I am not sure about the number of samples in the output file. I did have compared the output file size using the test dataset (chr22 only). 
`output=mt13.select_rows(mt13.rsid, mt13.qual)
output=output.select_entries(output.GT, output.DP, output.AD, output.GQ)
mt_fileout="chr22_QCed.mt"
output.write(mt_fileout, overwrite=False)`

The output file is pVCF (41KB) vs MT (1.6MB). Since pVCF seems to be much smaller, I decided to write pVCF. I am happy to write out mt files if this is preferred!